### PR TITLE
Fix transitive dependencies not being included

### PR DESCRIPTION
**Pull request by hdgarrood**
There was a bug where resolvedDependencies was only including top level
dependencies and not transitive ones. This commit fixes it.

'top level dependencies' here means those which appear at the top level
of the 'dependencies' key of `bower list --json`, as a result of either
having being declared in bower.json or having been installed manually
using `bower install`.

### CODE REVIEW

**Comment by hdgarrood** [comment]
This only concerns psc-publish, and fixes missing links in Pursuit docs, like Maybe not being linked [here](http://new-pursuit.purescript.org/packages/purescript-arrays/0.4.0/docs/Data.Array#d:head)

File: psc-publish/Main.hs [hunk 1]
```diff
@@ -196,18 +197,35 @@ getResolvedDependencies :: [PackageName] -> PrepareM [(PackageName, Version)]
...
+--   `bower list --json --offline`
+asResolvedDependencies :: Parse BowerError [(PackageName, DependencyStatus)]
+asResolvedDependencies = nubBy ((==) `on` fst) <$> go
+  where
+  go =
+    fmap (fromMaybe []) $
+      keyMay "dependencies" $
+        (++) <$> eachInObjectWithKey (parsePackageName . T.unpack)
+                                     asDependencyStatus
+             <*> (concatMap snd <$> eachInObject asResolvedDependencies)
```
Comment (paf31) [inline psc-publish/Main.hs]: If there is a cyclic dependency (unlikely but possible, I think), won't this `<<loop>>`?

Comment (hdgarrood) [inline psc-publish/Main.hs]: This is parsing the output of 'bower list --json --offline' so I think that would have to produce an infinite amount of output for this to cause an infinite loop. In such a case we probably wouldn't even get as far as this line; I think readProcess would never return. I will check this though.

Comment (hdgarrood) [inline psc-publish/Main.hs]: I just checked this - if you create a bower package with cyclic dependencies and then run `bower list --json --offline`,  it prints a dependency tree where packages are not listed where they would appear as children of themselves. So if A depends on B depends on A, the dependency tree will just look like `A -> B`. So I am pretty sure this is fine.

**Comment by paf31** [comment]
:+1: Thanks!

**Pull request merged**